Neuro/Minigames/MinigameHandler.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections;
+using BepInEx.Unity.IL2CPP.Utils;
+using Neuro.Minigames.Completion;
+using Neuro.Utilities;
+using Reactor.Utilities.Attributes;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace Neuro.Minigames;
+
+public static class MinigameHandler
+{
+    public static void CompleteMinigame(Minigame minigame, PlayerTask task)
+    {
+        // TODO: Make sure this works with the safe minigame on airship
+        // TODO: Make sure this works with the simon says minigame on skeld
+
+        if (minigame.TryCast<IDoorMinigame>() is { })
+        {
+            minigame.StartCoroutine(CompleteDoorMinigame(minigame));
+            return;
+        }
+
+        // If task is null, then the minigame can be one of the following:
+        // - Door minigame (handled above)
+        // - Spawn in minigame
+        // - Role ability minigame
+        // - System console minigame
+        if (task!?.TryCast<NormalPlayerTask>() is not { } normalPlayerTask) return;
```

Comment:
Couldn't this special logic be moved to `MinigameSolver#Complete` so `CompleteDoorMinigame` could be a normal MinigameSolver class?

Reply:
That's what the plan is

Reply:
Problem is that currently the solvers use TaskTypes to figure out which to instantiate, but door minigames dont have any task types.

Would probably need to change to using the minigame type directly